airbyte-workers/src/test/java/io/airbyte/workers/internal/StateMetricsTrackerTest.java

```diff
@@ -137,4 +138,31 @@ void testStateMetricsTrackerNoStateMatchExceptionThrown() throws StateMetricsTra
         () -> stateMetricsTracker.updateStates(s3.getState(), 4, LocalDateTime.parse(SECOND_FIVE, FORMATTER)));
   }
 
+  @Test
+  void testStreamMaxandMeanSecondsBeforeStateMessageEmitted() {
```

Comment:
What do you think of having this test call `handleSourceEmittedState` instead?
I think overall it is more valuable to ensure that this method is tracking as expected rather than extracting part of it in a test.

Reply:
yeah I think that's a good idea - it would mean moving everything into the AirbyteMessageTrackerTest. is it ok if we refactor this in a followup ticket?

Reply:
👍